app/hocs/withTranslation.tsx

```diff
@@ -0,0 +1,21 @@
+import type { ComponentType } from 'react'
+import type { Language } from '../hooks/useLanguage'
+import { useLanguage } from '../hooks/useLanguage'
+
+export type EventualString = string | ((...args: any[]) => string);
+export type Translation<T = Record<string, EventualString>> = Record<keyof T, EventualString>;
+export type Translations<T extends Translation> = Record<Language, T>;
+export type PropsWithTranslation<P, T extends Translation> = P & { translation: T };
```

Comment:
I think we should extend P like react itself with the `PropsWithChildren` type
`type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined };`
There is not always a need for props. And an `Empty` type in that case seams unnecessary.

Reply:
We could do `PropsWithTranslation<P = unknown, T extends Translation = Translation>` or change the order back around (I initially swapped it as I thought it would be more natural to have props come first)

Reply:
_Refactored!_